lib/service/Service.ts

```diff
@@ -58,6 +59,8 @@ const argChecks = {
   POSITIVE_AMOUNT: ({ amount }: { amount: number }) => { if (amount <= 0) throw errors.INVALID_ARGUMENT('amount must be greater than 0'); },
   POSITIVE_QUANTITY: ({ quantity }: { quantity: number }) => { if (quantity <= 0) throw errors.INVALID_ARGUMENT('quantity must be greater than 0'); },
   PRICE_NON_NEGATIVE: ({ price }: { price: number }) => { if (price < 0) throw errors.INVALID_ARGUMENT('price cannot be negative'); },
+  QUANTITY_DECIMAL_PLACES: ({ quantity }: { quantity: number }) => { if (!checkDecimalPlaces(quantity)) throw errors.INVALID_ARGUMENT('quantity mush have max 12 decimal places'); },
+  PRICE_DECIMAL_PLACES: ({ price }: { price: number }) => { if (!checkDecimalPlaces(price)) throw errors.INVALID_ARGUMENT('price mush have max 12 decimal places'); },
```

Comment:
```suggestion
  PRICE_DECIMAL_PLACES: ({ price }: { price: number }) => { if (!checkDecimalPlaces(price)) throw errors.INVALID_ARGUMENT('price must have less than 13 decimals); },
```

Maybe reword the error so that it is more clear?